### ISSUE TITLE
Chapter 2 feedback

### ISSUE THREAD

**Issue by Talon1024** [Realm667/WolfenDoom#171]
Please enclose the screenshots you post into spoilers, so the issue does not become a monstrosity like #178.
---

<details><summary>Original post by Talon1024 (entries not marked as fixed as of 25 Mar 2020 remain in the post's body [all checkboxes auto-filled on 29 Oct 2020 so as not to include them in the total count --N00b])</summary>

C2M1 - Operation: Ausbruch (Prison Break)
------------------

- [x] - I think it would be very helpful to add at least one more hint, such as "Avoid enemies' line of sight before moving around" to help players avoid being spotted unexpectedly.
- [x] - It may also be worth adding a hint that says "Not every area is worth scouring".
- [x] - If the alarm gets pulled on your ass, it's pretty much game over at that point. Should alarm panels be made breakable?
- [x] - This map might need the no-cone ceiling lights added in certain areas to improve visibility.
- [x] - One time, I awoke a sleeping guard by standing near him for too long. Is that supposed to happen?
- [x] - Somehow, I got stuck behind [this door](https://user-images.githubusercontent.com/6550500/65918545-e60a7780-e3a7-11e9-9840-83bf7ffbd45e.png). If I remember, I jumped down, killed the SS guys, and went for the secret treasure inside the vent before attempting to go back up and take on the mutant lab section.
- [x] - It's seemingly impossible to avoid being spotted after leaving the roundabout corridor to the cafeteria.
> Which place do you mean exactly @Talon1024 ? -- Ozy81

- [x] - The guard towers started screaming like crazy the moment they saw me walking through the light. I do not think they should be making any noise at all. Furthermore, I think they should be disabled once all the guards in that tower die.
- [x] - Overall, I have a lot of experience with this level, so it may be worth asking someone outside the team to test it out.

C2M2 - Operation: Feuerdrone (Sewer/Robot facility)
---------------------

- [x] - The Waffen SS and MP40 SS guards are really persistent with their shooting. This map has a lot of long corridors and other such areas where enemies attack you from medium to long range, and you have little cover, which makes combat really annoying on this map.
- [x] - Surely we can come up with a better visual effect for the drone's beam attack.

C2M3 - Operation: Schlangenkopf (Village/Castle)
---------------------

- [x] - It's weird that Douglas would warn you about snipers in front of the cathedral when he isn't even with you for this mission.
- [x] - Sometimes, doors can get blocked by things above or below them. I know how to fix this, though.
- [x] - Doors can also get blocked by enemies behind them. The houses in the village are a bit cramped.
- [x] - It's also weird that a huge spider web in the catacombs can block you. Maybe put some rocks in the way instead?
- [x] - The undead monks shout in German when you shoot at them.
- [x] - The missile-launching giant mutants can also be a huge issue.

C2M4 - Operation: Wustenfuchs (Desert/Bunker)
---------------------

- [x] - It is a bit difficult to aim at things with the Fliegerfaust (truck weapon) when you're constantly flinching because of the enemies firing at you.
- [x] - After arriving in Mahdia, some enemies that you left behind are still able to shoot at you through the gates.

- [x] Just before the end of the map, there is a glitch (player with Fliegerfaust is teleported to the right of the truck).

C2M5_A - Operation: Stahlhimmel (Train station)
---------------------

- [x] - This map DEFINITELY needs the no-cone ceiling lights inside the train cars. The ceiling light halos are making it really hard to see down the train car hallways.
- [x] - It's a bit tricky to get on the machinegun turret in time to stop the dogs and the ubersoldat from seriously injuring you.
- [x] - Should the secret area/push wall be more visible? All the other secrets in WolfenDoom stand out a bit more than that one. In case you don't know where it is, it's on the left side of the tunnel.
- [x] - Douglas was attacking one of the sniper nests before he called me down to get in the truck.

C2M5_B - Operation: Stahlhimmel (Rocket Facility)
---------------------

- [x] - This mission has the same problem as C2M2 when you first meet the MP40-toting elites... Long corridors, and little cover.
- [x] - Trying to open [this door](https://user-images.githubusercontent.com/6550500/65895622-bf354c80-e379-11e9-88f6-e6caf15b9c7d.png) does not work, unless you are pressing your body against it. I think that problem has something to do with the climbable zone right behind it. Speaking of which, I found myself able to fly at one point.
- [x] - I deviated from the expected route and found myself behind a [padlocked door](https://user-images.githubusercontent.com/6550500/65894342-9ca23400-e377-11e9-90a1-8d7de820693d.png). I think it would make more sense to make that side openable with the cyan key rather than forcing the player to go through the fan area again.
- [x] - It may help to increase the volume and radius of the power on/off sounds in the fan area. Maybe there should also be a switch that turns the fans off?
- [x] - The MG turret power switch should have a sign above it. I forgot what the German for "Machinegun turret power switch" is, unfortunately. @Tormentor667?

C2M0_A - Operation: Astrostein (Ancient Mayan Ruins)
---------------------

- [x] - I have a lot of experience with this mission, so it may be worth having someone outside the team look at this mission.
- [x] - Inexperienced players may need to be encouraged to look above to find the last switch they need to shoot in order to raise the last piece of the wooden bridge.

C2M0_B - Operation: Astrostein (SS Eisenherz/New Berlin moon base)
---------------------

- [x] - I have a lot of experience with this map, but I think it would look very confusing for someone who is playing this level for the first time. I remember having to tell [Samsai](https://twitch.tv/sirsamsai) where to go in order to get the red key (behind the strippers in the strip club).
- [x] - The fight near the yellow key is still a bit annoying, especially with the explosive canisters all around the field.
- [x] - Somehow, [this happened](https://user-images.githubusercontent.com/6550500/65894466-d7a46780-e377-11e9-960f-13754e027a35.png) after I left the bar.

C2M6_A - Operation: Hades (Bergen, Norway base)
---------------------

- [x] - At the start of this mission, you face a lot of enemies with rifles. The tesla shield helps, but it still doesn't last very long.
- [x] - Inside the house which is locked by the blue key, there is a 3D furnace which is facing the wrong way.
- [x] - This map has a few places where snipers can see you through the fog, and you can't see them, such as when you're approaching the dual gate, and at the train station where you board the train to Trondheim.
- [x] - The room with the crates containing the green key has some misaligned textures.
> Sadly this still depends by Worldpanning issue --Ozy81

C2M6_B - Operation: Hades (Trondheim, Norway base/Train Station Sneaky beaky level)
---------------------

- [x] - I do not know if it is possible to avoid a gunfight with the nazis at the train station after leaving the sewer, but I failed to do so. As a result, it caused a lot of the nazis on the other side of the map (the main area, after collecting the blue key) to move around a bit.
- [x] - One particular Gestapo officer was blocking an important switch that is used to open the way out of the volcano lab. I opened fire, but thankfully, that didn't alert everyone in the base. This may have been the result of the previous point, but I do not know.
- [x] - The previous notes were from my first run. I managed to avoid alerting the patrolling guards before they could raise an alarm, but I had to use a grenade to clean up the rest in the blue key room.
- [x] - The [sharks here](https://user-images.githubusercontent.com/6550500/65894536-f276dc00-e377-11e9-9f6a-16d4d9b0276f.png) pose almost no threat. They don't attack you or get blown up by the depth charges.
- [x] - The door repair script needs a small fix, since Douglas messages you AFTER you've fixed the door once you've found the repair kit.

C2M6_C - Operation: Hades (Reykjavik, Iceland base/Volcano base)
---------------------

- [x] - The [climbable zone here](https://user-images.githubusercontent.com/6550500/65894568-015d8e80-e378-11e9-80a9-faba680312e9.png) doesn't seem to work.
- [x] - Should the player get another armor once he enters the volcano base, or enters the mines? I think it would be appropriate to give the player a set of heavy armor there, or at least before the Angel of Death boss fight.
- [x] - The letter you read in Douglas' bedroom at the start of Chapter 3 implies that he knew all about the fact James Ryan was a traitor. Should Douglas mention that, or hint at that to the player before we see Douglas get murdered?
- [x] - The timing on [this](https://user-images.githubusercontent.com/6550500/65895682-d7a56700-e379-11e9-9826-d25685279f93.png) [door](https://user-images.githubusercontent.com/6550500/65895683-d7a56700-e379-11e9-9851-2d9f08834186.png) is really tight. Should we give the player more time to go through the door?

General feedback
----------------

- [x] - I think static Wehrmacht riflemen are a bit OP, since there is no delay between when they see you, and when they open fire. The riflemen are also the equivalent of Revenants IMO.
- [x] - I think the flinching from the big rocket-launching mutant's poison clouds should be reduced or removed.
- [x] - It seems like enemies with flamethrowers are very underpowered now. They aren't even as threatening as pistol guards, unless they decide to explode in your face. Speaking of which, there are quite a few times you have to face them at close range, such as in C2M4, or in C2M6_A, so should their death explosion splash damage and radius be reduced?
- [x] - The "sprite shadows" are not following the actors they should be for some reason.
- [x] - The game seems to be autosaving at the most inconvenient of times. Every time the game does that, it stalls for a split second. I ended up turning autosave off.
- [x] - The barrels need to have their normals fixed.
</details>

<details><summary>Resolved issues</summary>

**General:**

- [x] - The game seems to be autosaving at the most inconvenient of times. Every time the game does that, it stalls for a split second. I ended up turning autosave off.

- [x] After getting on and off a ladder the companions float mid-air (seen with Ryan in the central prototype room of C2M2):

<details><summary>Screenshots</summary>

![Screenshot_Doom_20200820_163719](https://user-images.githubusercontent.com/59409713/91416414-082aa500-e858-11ea-9299-22d3d8c74c3d.png)
![Screenshot_Doom_20200820_164640](https://user-images.githubusercontent.com/59409713/91416450-12e53a00-e858-11ea-8540-bf1933c6e2b2.png)
</details>

**C2 INTERMAP:**

**C2M1:**

- [x] In the outdoors area, (1667, -697, 160) the guards who are patrolling it do not notice the player when he is crouching on the passages.

> Which ones? Sleeping guards and snipers are stealth sensible, so that's intentional. On the other hand, enemies tagged as 219 gets removed after the introduction sequence, and some of them are also the Officer and the dobermann in front of the gates

> There were some ones (with MP 40, I guess?) who walked between the sniper towers, they did not see me even face to face. --N00b

**C2M2:**

- [x] This and the symmetrical subtractive dynamic light look silly. Maybe it would be better to lower the surrounding sectors' brightness to hide the snipers?

<details><summary>Screenshot</summary>

![Screenshot_Doom_20200307_215237](https://user-images.githubusercontent.com/59409713/76151461-60ad9080-60c6-11ea-8037-e38bdcd348d9.png)

</details>

- [x] (video) Dynamic lights do not work on moving floors for some reason:

[c2m2_1.zip](https://github.com/Realm667/WolfenDoom/files/4302472/c2m2_1.zip)
> Not sure what do you mean here since the video is very bright and if it is a matter of spotlights afaik they shouldn't move with sectors, but there where lots of Dormant/Ambush flags set even over lights which didn't had tags so maybe my recent commit should have fix it. --Ozy81

:x:  The Mechas are quite irritating with their >50-damage high-velocity rockets. An idea on rebalancing them: make them quickly (in 2–3 seconds) shoot out a large barrage (say 20 projectiles) of slower-moving rockets with low damage (like 10 or 15) into a wide cone (like ±15 degrees), and their trajectories should be non linear (for example, a helix + some random changes in velocity direction). This will add to the 'wacky tech' impression left by the machine and will make it fun to play against and much less frustrating to deal with.

<details><summary>Screenshot</summary>

![Screenshot_Doom_20200307_215835](https://user-images.githubusercontent.com/59409713/76151468-65724480-60c6-11ea-8116-a880e6469364.png)

</details>

**C2M3:**

- [x] - Sometimes, doors can get blocked by things above or below them. I know how to fix this, though.

- [x] - Doors can also get blocked by enemies behind them. The houses in the village are a bit cramped.

- [x] This polyobject door blocks players who try to pass under it.

<details><summary>Screenshot</summary>

![Screenshot_Doom_20200320_133659](https://user-images.githubusercontent.com/59409713/77157890-51a9e380-6ab3-11ea-80c4-f97b25a31b05.png)

</details>

**C2M4:**

- [x] Some of the enemies spawn directly in front of the truck (near (5866, -2900, 132) ). There are also Wehrmacht flamer operators spawned.

> They doesn't get spawned, they are intentionally on map, even flamers... Not sure what do you mean then :question:  --Ozy81

> CNR --N00b

- [x] Fliegerfaust has 'infinite ammo', so its ammotype should probably be cleared (now it just shows 'zero mini-rockets'.

**C2M5_A:**

**C2M5_B:**

- [x] The platforming in area at (5571, 12508, 548) with fans and UMG43 is (and was) too tough for my tastes; maybe lower the wind speed a little, lower the damage in the pit and allow the player to return to the start of the area (using a ladder)?

- [x] The first of the mechas (forgot to write down coordinates...) does not get occupied, it just stands there until the end of the level.

- [x] - This mission has the same problem as C2M2 when you first meet the MP40-toting elites... Long corridors, and little cover.

- [x] - I deviated from the expected route and found myself behind a [padlocked door](https://user-images.githubusercontent.com/6550500/65894342-9ca23400-e377-11e9-90a1-8d7de820693d.png). I think it would make more sense to make that side openable with the cyan key rather than forcing the player to go through the fan area again.

- [x] - It may help to increase the volume and radius of the power on/off sounds in the fan area. Maybe there should also be a switch that turns the fans off?

- [x] - The MG turret power switch should have a sign above it. I forgot what the German for "Machinegun turret power switch" is, unfortunately. @Tormentor667?

**C2M0_A:**

- [x] - Inexperienced players may need to be encouraged to look above to find the last switch they need to shoot in order to raise the last piece of the wooden bridge.

**C2M0_B:**

**C2M6_A:**

- [x] - Inside the house which is locked by the blue key, there is a 3D furnace which is facing the wrong way.

- [x] - This map has a few places where snipers can see you through the fog, and you can't see them, such as when you're approaching the dual gate, and at the train station where you board the train to Trondheim.

- [x] Too many paratroopers spawn after a certain moment in the level. They are tough, damaging, fast and spawn in packs, and considering the health shortage (even the health secret contains only 36%) this part may become very frustrating. I would spread them more evenly and cut their numbers by about one third.

- [x] "Sorry, restroom out of order" in a V2 complex at (1510, 15619, -143) (intentional?)

- [x] - At the start of this mission, you face a lot of enemies with rifles. The tesla shield helps, but it still doesn't last very long.

**C2M6_B:**

- [x] - I do not know if it is possible to avoid a gunfight with the nazis at the train station after leaving the sewer, but I failed to do so. As a result, it caused a lot of the nazis on the other side of the map (the main area, after collecting the blue key) to move around a bit.

- [x] - One particular Gestapo officer was blocking an important switch that is used to open the way out of the volcano lab. I opened fire, but thankfully, that didn't alert everyone in the base. This may have been the result of the previous point, but I do not know.

- [x] - The previous notes were from my first run. I managed to avoid alerting the patrolling guards before they could raise an alarm, but I had to use a grenade to clean up the rest in the blue key room.

- [x] - The door repair script needs a small fix, since Douglas messages you AFTER you've fixed the door once you've found the repair kit.

**C2M6_C:**

</details>

General
---

C2 INTERMAP
---

C2M1 - Operation: Ausbruch (Prison Break)
---

C2M2 - Operation: Feuerdrone (Sewer/Robot facility)
---

C2M3 - Operation: Schlangenkopf (Village/Castle)
---

C2M4 - Operation: Wustenfuchs (Desert/Bunker)
---

- [x] You can fire the Fliegerfaust 'right into the truck' (below yourself), the possible pitch difference needs to be bounded (BTW, is roll changed while the truck is moving?). _[assigned to @AFADoomer and @Talon1024]_

C2M5_A - Operation: Stahlhimmel (Train station)
---

C2M5_B - Operation: Stahlhimmel (Rocket Facility)
---

C2M0_A - Operation: Astrostein (Ancient Mayan Ruins)
---

C2M0_B - Operation: Astrostein (SS Eisenherz/New Berlin moon base)
---

C2M6_A - Operation: Hades (Bergen, Norway base)
---

C2M6_B - Operation: Hades (Trondheim, Norway base/Train Station Sneaky beaky level)
---

C2M6_C - Operation: Hades (Reykjavik, Iceland base/Volcano base)
---

**Comment by Tormentor667** [comment]
Very good stuff! Let me take care of that :)

**Comment by Ozymandias81** [comment]
I will check carefully what needs to be done and what I can fix this week, and once finished you'll go through them then we'll see what's left... What do you think @Tormentor667 ? Btw the destryable alarm panels was something I proposed in the past, and now it pops up from Talon and not me :)

**Comment by Talon1024** [comment]
@Ozymandias81 I think it's line 1590 of [base.txt](https://github.com/Realm667/WolfenDoom/blob/master/scripts/actors/enemies/base.txt#L1590) which is causing the undead monks to shout in German.

**Comment by Ozymandias81** [comment]
I see, gonna give it a go

**Comment by Ozymandias81** [comment]
So finally I am doing some severe progress on [fixing C2M1](https://github.com/Realm667/WolfenDoom/commit/dea1df1277ebfcd35d96ee5b90fdb8e85deff570) , I also forgot to say that for sleeping soldiers I simply rotated their angle in front of walls backfacing the courtyard, so we must alert monsters to make them wake up.
Beside alarm panels being destroyable, quite all you have mentioned for C2M1 has been fixed (for the cafeteria I plan to add some ammo... But to be sure, what is the cafeteria you mean?) @Talon1024 , I have also spot more problems which I should take care in a couple of commits

**Comment by AFADoomer** [comment]
I implemented destroyable alarm panels last night, but apparently forgot to commit the changes back to the repo...  Will push the button on those tonight.

**Comment by Ozymandias81** [comment]
Good, hope you didn't edit the map though! I have spotted plenty problems with c2m1 and fixing quite everything, in case the actor need some updates on the map simply add that on your test map and I will check if something needs to be updated :)

**Comment by AFADoomer** [comment]
No, no map changes...  Just changes to the Alarm Panel actor that will work as-is with the current map placements.

**Comment by Ozymandias81** [comment]
Thanks again for that update, then new actor now makes more sense. Fantastic patches for it!
[Fixed](https://github.com/Realm667/WolfenDoom/commit/b357c14fee392d7ba59a860bb5943e6310146484) something I have noticed with AlarmPanel and also went through more deeply with c2m1, there are still a couple of things to take care but after that finally will be time for c2m2

**Comment by Tormentor667** [comment]
...now here is my feedback regarding Kevin's Chapter 2 review:

**C2M1**
I think @Ozymandias81 is already working on most of the stuff being said, so there is no need for me to expand upon that.

**C2M2**
Regarding the Waffen SS and MP40 guards, I am okay with any changes necessary, who will take care of both points? And what about a good drone effect?

**C2M3**
I fixed the message regarding Douglas, makes more sense now. The monk sound is fixed by @Ozymandias81. Also fixed the non-passable webs. Can you take care of the blocking doors @Talon1024? And regarding the missile-giant mutants, what suggestion do you have for them?

**C2M4**
I'd suggest reducing the flinch amount when sitting on the truck (maybe by half). Is something like that possible @Talon1024 and @AFADoomer? I fixed the remaining enemies in front of the gates after you arrived Mahdia (removing them). Also fixed a visual issue when driving through the canyon.

**C2M5_A**
@Ozymandias81 will you take care of the no-cone lights? I moved the turret farther away from the ubersoldat and the dogs to have more time to reach it. Made the secret nearby stand out a bit more. Regarding Douglas attacking a sniper nest: also fixed.

**C2M5_B**
Regarding the MP40s and Waffen SS units, I'd suggest you fix it the same way you do it in C2M2 @Talon1024. I hopefully fixed the door/climbable zone issue you posted. Made the door openable from the other side with a cyan key. I am okay with increasing the menue and adding a switch to deactivate the fans where it makes sense. You seem to have an idea already so I will leave this up to you @Talon1024. Also added a sign to the button for the turret.

**C2M0_A**
Added a small sign to the bridge puzzle to hint people to look above.

**C2M0_B**
Added signs as a hint to enter the backstage area of the stripping stage. Regarding the boss battle, @Talon1024 do you have a good idea to improve it and make it entertaining and challenging? Regarding the screenshot you've posted, this is the responsible script:
`script 20 (void)
{
	thing_spawnfacing(225, "AstroElite", 0, 0);
}`
Scriptwise it does exactly what it is supposed to do. Spawning "AstroElite" through the console works also. No idea what is going on with the script in your special case though. Any ideas?

**C2M6_A**
Fixed the furnace at the blue-key-locked house, I don't find the misalignment at the green key though. Regarding the beginning of the map, I think the difficulty is just right, you simply need to act slow and precise to survive the enemies and the snipers. No idea if that needs a real change.

**C2M6_B**
Added a few more sound/sight blocking lines to ensure that not everybody is awake after the first kill past the sewers. Regarding the sharks: They should attack you, I wonder why they don't. Can anyone fix that? @Talon1024 can you fix the door script message? I am not sure atm what's the right change to make the sequence make sense.

**C2M6_C**
Fixed the climbable zone. The door sequence has more delay near the end. Added an armor near the final area. Regarding the hint of Ryan's treason: I'd leave it as it is, because it makes the letter at the beginning of C3 so much stronger.

**General feedback**
I agree with everything and would appreciate it if you can take care of the changes. Everything I mentioned above has been worked and fixed so far, check my commit from just now for details :) Other than that: Good work on your playthrough!

**Comment by AFADoomer** [comment]
Added script-controllable player flinch scaling in dd9cadcd4a92a5c02d1133839eb9596d292e36b3.  Also fixed up the C2M4 truck sequence script to reduce flinching by half while you are on the truck.

Also going to do a pull request for proposed alternate long-range fire for ProtoDrone.

**Comment by Tormentor667** [comment]
@Username-N00b-is-not-available can you also share screenshots with coordinates to the places in question?

- General: The game seems to be autosaving at the most inconvenient of times. Every time the game does that, it stalls for a split second. I ended up turning autosave off.
- C2M3: Sometimes, doors can get blocked by things above or below them. I know how to fix this, though.
- C2M3: Doors can also get blocked by enemies behind them. The houses in the village are a bit cramped.
- C2M4: Just before the end of the map, there is a glitch (player with Fliegerfaust is teleported to the right of the truck).
- C2M6_B: One particular Gestapo officer was blocking an important switch that is used to open the way out of the volcano lab. I opened fire, but thankfully, that didn't alert everyone in the base. This may have been the result of the previous point, but I do not know. **T667: BTW, have you considered stabbing him with your knife?**

Some questions regarding issues you mentioned:
- C2M5_B: It may help to increase the volume and radius of the power on/off sounds in the fan area. Maybe there should also be a switch that turns the fans off? **T667: It's already played at max, where do you want to place the switch that turns off the fans?**
- C2M6_B: The door repair script needs a small fix, since Douglas messages you AFTER you've fixed the door once you've found the repair kit. **T667: Can you tell me where the place is where you encountered the wrong message?**

**Comment by Talon1024** [comment]
Some of these issues have been fixed. I really haven't been checking these threads on GitHub, since they tend to get long and complicated.

As for the issues that (AFAIK) have been fixed:
- C2M6_B: One particular Gestapo officer was blocking an important switch that is used to open the way out of the volcano lab. I opened fire, but thankfully, that didn't alert everyone in the base. This may have been the result of the previous point, but I do not know.
- C2M6_B: The door repair script needs a small fix, since Douglas messages you AFTER you've fixed the door once you've found the repair kit.

**Comment by Username-N00b-is-not-available** [comment]
The problems which start with a hyphen after the checkbox are from Talon1024's original post, I just preserved them as is.

**Comment by Username-N00b-is-not-available** [comment]
@Tormentor667 @Ozymandias81 @AFADoomer @Talon1024, nice work regarding the fixes in first two chapters! I would just like to tell you that I made a slightly improved version of CUTSCEN2 (as well as some other fixes not connected to maps), and when I'll regain a stable internet connection, most likely a week later, I'll commit these changes.

**Comment by Ozymandias81** [comment]
Sounds cool for me man @Username-N00b-is-not-available , I am curious what are about these fixes you mentioned which doesn't regard maps? Just to have an idea. :)

**Comment by Username-N00b-is-not-available** [comment]
@Ozymandias81 There's not much, but I'd name adding scopes to enemy sniper sprites. What do you think, does the scope on the image fit or is it too cartoonish?
![SNIWC0](https://user-images.githubusercontent.com/59409713/90419952-f90d5f80-e0bf-11ea-800f-2e590155377a.png)

**Comment by Tormentor667** [comment]
I like that, but remember that this also needs to be added to all rotations.

**Comment by Tormentor667** [comment]
> I would just like to tell you that I made a slightly improved version of CUTSCEN2 (as well as some other fixes not connected to maps), and when I regain a stable internet connection, most likely a week later, I'll commit these changes.

What did you change? :)

**Comment by Username-N00b-is-not-available** [comment]
> this also needs to be added to all rotations

Yeah, I've already done all the rotations (I used the Kar98k pickup sprite's scope for other ones). What now remains is to upload the sprites and crush them (and also change sprite name in DECORATE for one of the snipers, as it is shared with regular riflemen).
I hope I did not miss any snipers :-) (I've done Wehrmacht, Afrika Korps and the winter ones).

>What did you change?

Added camera movement to every scene, made a second floor in the ammo factory and redesigned it a bit, put a small station (its building is copied from C1M0) near the rail tracks. I'll think more of what to do with it, though.

**Comment by Ozymandias81** [comment]
Looks cool, will look forward these additions.

**Comment by Tormentor667** [comment]
Same here :)

**Comment by Username-N00b-is-not-available** [comment]
Also working on an Afrika Korps Flammenwerfer soldier recolor, to be used in C2M4 (and possibly other desert maps, does any of them include this type of soldier?)
![FLS2D1](https://user-images.githubusercontent.com/59409713/90678343-3f50f300-e267-11ea-8355-507931699afb.png)

**Comment by Tormentor667** [comment]
Only C2M4 afaik, maybe C2M0_A could benefit from them

**Comment by Tormentor667** [comment]
> The game seems to be autosaving at the most inconvenient of times. Every time the game does that, it stalls for a split second. I ended up turning autosave off.

@Username-N00b-is-not-available do you think you can manually change the places which seem to be inconvenient for you? It's hard to fix that without knowing which places are meant to be improved :) 

> The Mechas are quite irritating with their >50-damage high-velocity rockets. An idea on rebalancing them: make them quickly (in 2–3 seconds) shoot out a large barrage (say 20 projectiles) of slower-moving rockets with low damage (like 10 or 15) into a wide cone (like ±15 degrees), and their trajectories should be non linear (for example, a helix + some random changes in velocity direction). This will add to the 'wacky tech' impression left by the machine and will make it fun to play against and much less frustrating to deal with.

@Username-N00b-is-not-available I think I need to see that in action. It sounds as it would work well in open areas like C2M2 (arena) and also C2M5_B (tunnels) but I am not sure if it works in tight situations like the C2M2 escape sequence or one area in C3M1_B. What do you think? As of now, I think it works well in large spaces, you can strafe-escape the rockets quite good if you stay in movement.

> You can fire the Fliegerfaust (C2M3) 'right into the truck' (below yourself), the possible pitch difference needs to be bounded (BTW, is roll changed while the truck is moving?).

@AFADoomer maybe this is something you can take care of?

> Just before the end of the map, there is a glitch (player with Fliegerfaust is teleported to the right of the truck).

@Username-N00b-is-not-available can you make a video of the problem? I can't reproduce this. For me the map ends correctly.

**Comment by Username-N00b-is-not-available** [comment]
@Tormentor667: 
1. https://github.com/Realm667/WolfenDoom/issues/171#issuecomment-665078301 **I have no objections** to savegame placement. This checkbox is a leftover of the **almost year-old** original post. Please ask @Talon1024 if he still thinks that this should be changed.
2. Yes, over time I've changed my opinion on this. The checkbox should be discarded.
4. Already done :-) C2, part XII, around 11:26.
